paimon-core/src/main/java/org/apache/paimon/iceberg/migrate/IcebergMigrator.java

```diff
@@ -0,0 +1,393 @@
+                            .collect(Collectors.toList());
+
+            // Again, check if delete File exists
+            checkAndFilterDataFiles(icebergDataFileMetas);
+
+            LOG.info(
+                    "Begin to create Migrate Task, the number of iceberg data files is {}",
+                    icebergDataFileMetas.size());
+
+            List<MigrateTask> tasks = new ArrayList<>();
+            Map<Path, Path> rollback = new ConcurrentHashMap<>();
+            if (paimonTable.partitionKeys().isEmpty()) {
+                tasks.add(importUnPartitionedTable(icebergDataFileMetas, paimonTable, rollback));
+            } else {
+                tasks.addAll(importPartitionedTable(icebergDataFileMetas, paimonTable, rollback));
+            }
+
+            List<Future<CommitMessage>> futures =
+                    tasks.stream().map(executor::submit).collect(Collectors.toList());
+            List<CommitMessage> commitMessages = new ArrayList<>();
+            try {
+                for (Future<CommitMessage> future : futures) {
+                    commitMessages.add(future.get());
+                }
+            } catch (Exception e) {
+                futures.forEach(f -> f.cancel(true));
+                for (Future<?> future : futures) {
+                    // wait all task cancelled or finished
+                    while (!future.isDone()) {
+                        //noinspection BusyWait
+                        Thread.sleep(100);
+                    }
+                }
+                // roll back all renamed path
+                for (Map.Entry<Path, Path> entry : rollback.entrySet()) {
+                    Path newPath = entry.getKey();
+                    Path origin = entry.getValue();
+                    if (paimonFileIO.exists(newPath)) {
+                        paimonFileIO.rename(newPath, origin);
+                    }
+                }
+
+                throw new RuntimeException("Migrating failed because exception happens", e);
+            }
+            try (BatchTableCommit commit = paimonTable.newBatchWriteBuilder().newCommit()) {
+                commit.commit(new ArrayList<>(commitMessages));
+                LOG.info("paimon commit success! Iceberg data files has been migrated to paimon.");
+            }
+        } catch (Exception e) {
+            paimonCatalog.dropTable(paimonIdentifier, true);
+            throw new RuntimeException("Migrating failed", e);
+        }
+
+        // if all success, drop the origin table according the delete field
+        if (deleteOriginTable) {
+            icebergMigrateMetadata.deleteOriginTable();
+        }
+    }
+
+    @Override
+    public void deleteOriginTable(boolean delete) throws Exception {
+        this.deleteOriginTable = delete;
+    }
+
+    @Override
+    public void renameTable(boolean ignoreIfNotExists) throws Exception {
+        LOG.info("Last step: rename.");
+        LOG.info("Iceberg migrator do not rename table now.");
```

Comment:
No need for this log. Also why Iceberg migrator ignore this method? Add comments.